Update app signing introduction (#9700)
flutter/website: src/deployment/android.md

+<style name="NormalTheme" parent="Theme.MaterialComponents.DayNight.NoActionBar">
```

> ## Sign the app

Reviewer: Since this page has existed for a while, there may be links to the old fragment. You can add a backup anchor for the old links like this:

```suggestion
<a id="signing-the-app"></a>
## Sign the app
```

> To publish on the Play Store, you sign your app with a digital signature.

Reviewer: I think "need" is important here as you need to do more than just sign the app to publish it. Also use "certificate" here to more closely align with the Android docs:

```suggestion
To publish on the Play Store, you need to
sign your app with a digital certificate.
```

Android uses two signing keys: _upload_ and _app signing_.

> * Developers upload an `.aab` or `.apk` file signed with an _upload key_ to the Play Store.

Reviewer: ```suggestion
* Developers upload an `.aab` or `.apk` file signed with
  an _upload key_ to the Play Store.
```

* The end-users download the `.apk` file signed with an _app signing key_.

To create your app signing key, use Play App Signing

> as described in the [official Play Store documentation][].

Reviewer: I'm a bit confused here as this text says to use Play App Signing to create my key, then the next category again gives me two options to create my key including that one. Perhaps this note can be dropped?

Author: You need to create the upload key and keystore first, then you can use the Play App Signing. This does explain that.
